Test did not account for non existing fields in some records

Considering using test data below:
```
{"docType":"Account","accountNo":"li_test_1","customerNo":"236","accountStatus":"1","accountTradingStopFlag":"false","balance":14000,"createdAt":1539914728331,"updatedAt":1539914728332}
{"docType":"Account","accountNo":"li_test_2","customerNo":"236","accountStatus":"1","accountTradingStopFlag":"false","balance":26000,"createdAt":1539914728331,"updatedAt":1539914728332}

{"docType":"TransactionInfo","iribaraiId":"ut-001_shiharai","tradingNo":"ut-001","accountNo":"li_test_1","transactionKubun":"shiharai","cancelKubun":"","tradingDate":1539914728333,"tradingAmount":3000,"transferFromTo":"li_test_2","remarks":"remarks","acceptedAt":1539914728333,"tradingStatus":"1","createdAt":1539914728333,"updatedAt":1539914728333,"serverId":"server001","sentAt":1539914728333,"detailsLink":"details link 123"}
{"docType":"TransactionInfo","iribaraiId":"ut-001_uketori","tradingNo":"ut-001","accountNo":"li_test_2","transactionKubun":"uketori","cancelKubun":"","tradingDate":1539914728333,"tradingAmount":3000,"transferFromTo":"li_test_1","remarks":"remarks","acceptedAt":1539914728333,"tradingStatus":"1","createdAt":1539914728333,"updatedAt":1539914728333,"serverId":"server001","sentAt":1539914728333,"detailsLink":"details link 123"}
```
This is the query string:
```
{"selector":{"docType":"TransactionInfo","serverId":{"$ne":"server002"},"tradingDate":{"$gt":1539914728330}}}
```

When running the test i got this error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x98 pc=0x975b4e]

goroutine 26 [running]:
testing.tRunner.func1(0xc0000cca00)
	C:/Go/src/testing/testing.go:792 +0x38e
panic(0xa6f600, 0x10c4de0)
	C:/Go/src/runtime/panic.go:513 +0x1c7
github.com/wearetheledger/go-couchdb-query-engine/query/rules.eq.Match(0xb387ff, 0x3, 0x0, 0x0, 0xa2b440, 0xc0003f84b0, 0x3, 0xffffffffffffff01, 0x0)
	C:/Users/aaa/go/src/github.com/wearetheledger/go-couchdb-query-engine/query/rules/eq.go:11 +0x8e
github.com/wearetheledger/go-couchdb-query-engine/query/rules.ne.Match(0xb3880b, 0x3, 0x0, 0x0, 0xa2b440, 0xc0003f84b0, 0x40ed59, 0xc000401ee8, 0xc0001c17d0)
	C:/Users/aaa/go/src/github.com/wearetheledger/go-couchdb-query-engine/query/rules/ne.go:8 +0x6f
github.com/wearetheledger/go-couchdb-query-engine/query.test(0x0, 0x0, 0xa59220, 0xc0001c17d0, 0x10f8d60, 0xc0003f8520, 0x1)
	C:/Users/aaa/go/src/github.com/wearetheledger/go-couchdb-query-engine/query/query.go:174 +0x337
github.com/wearetheledger/go-couchdb-query-engine/query.test(0xa59220, 0xc0001c1740, 0xa59220, 0xc0001c17a0, 0xc0000c0aa8, 0x0, 0x0)
	C:/Users/aaa/go/src/github.com/wearetheledger/go-couchdb-query-engine/query/query.go:190 +0x2b6
github.com/wearetheledger/go-couchdb-query-engine/query.ParseCouchDBQueryString(0xc000402210, 0xc0000dc1c0, 0x6a, 0x9, 0xc0001d4338, 0x0, 0x0, 0xc0001da000)
	C:/Users/aaa/go/src/github.com/wearetheledger/go-couchdb-query-engine/query/query.go:94 +0x2de
github.com/hyperledger/fabric/core/chaincode/shim.(*CustomMockStub).GetQueryResult(0xc000096190, 0xc0000dc1c0, 0x6a, 0x96, 0x0, 0x0, 0x95)
	C:/Users/aaa/go/src/github.com/hyperledger/fabric/core/chaincode/shim/custom_mockstub.go:38 +0x3a1
ccc.(*JcnCashlessAsset).findTransactionsByServerId(0x10f7d28, 0xc01ea0, 0xc000096190, 0xc00006b010, 0x3, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
	C:/Users/aaa/Documents/bbb/workspace/ccc/src/ccc/ccc.go:92 +0x57e
ccc.(*JcnCashlessAsset).Invoke(0x10f7d28, 0xc01ea0, 0xc000096190, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	C:/Users/aaa/Documents/bbb/workspace/ccc/src/ccc/ccc.go:71 +0xd5c
github.com/hyperledger/fabric/core/chaincode/shim.(*CustomMockStub).MockInvoke(0xc000096190, 0xc000459cb0, 0x24, 0xc0000a8a20, 0x4, 0x4, 0x0, 0x0, 0x0, 0x0, ...)
	C:/Users/aaa/go/src/github.com/hyperledger/fabric/core/chaincode/shim/custom_mockstub.go:51 +0xe7
ccc.TestFindTransactionsByServerId(0xc0000cca00)
	C:/Users/aaa/Documents/bbb/workspace/ccc/src/ccc/ccc_test.go:129 +0xd9f
testing.tRunner(0xc0000cca00, 0xb67e90)
	C:/Go/src/testing/testing.go:827 +0xc6
created by testing.(*T).Run
	C:/Go/src/testing/testing.go:878 +0x35a
FAIL	ccc	1.401s
```

The reason is because the test data(docType is Account) doesn't have a field property called [serverId]. so the data being passed to test function in query.go is nil.

I fixed this issue and made a pull request.

By the way,in the loop below in test function of query.go
Once one of the query result is false, there is no need to check the others.
So i checked the value of last(boolean flag), if it is false, i think we can return false right away.
This way makes the performance better.
```
for k, v := range userQuery {
.........
   // i added this code
   if !last {
      return last, nil
   }
}
```

Please check my pull request. Thank you very much. hope this package can be used in MockStub soon.

--- a/query/query.go
+++ b/query/query.go
@@ -168,7 +168,9 @@ func test(data interface{}, userQuery interface{}) (bool, error) {
 				if err != nil {
 					return false, err
 				}
-
+				if rule.GetName() != "$exists" && data == nil {
+					return false, nil
+				}
 				matched, err1 := rule.Match(data, v)
 
 				if err1 != nil {
@@ -197,6 +199,9 @@ func test(data interface{}, userQuery interface{}) (bool, error) {
 
 				last = last && result
 
+			}
+			if !last {
+				return false, nil
 			}
 		}
 

